ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/DLModelExecute.java

```diff
@@ -0,0 +1,233 @@
+package org.opensearch.ml.engine.algorithms;
+
+import ai.djl.Application;
+import ai.djl.Device;
+import ai.djl.engine.Engine;
+import ai.djl.inference.Predictor;
+import ai.djl.repository.zoo.Criteria;
+import ai.djl.repository.zoo.ZooModel;
+import ai.djl.translate.TranslateException;
+import ai.djl.translate.Translator;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.io.FileUtils;
+import org.opensearch.ml.common.FunctionName;
+import org.opensearch.ml.common.MLModel;
+import org.opensearch.ml.common.exception.MLException;
+import org.opensearch.ml.common.input.Input;
+import org.opensearch.ml.common.model.MLModelConfig;
+import org.opensearch.ml.common.model.MLModelFormat;
+import org.opensearch.ml.common.output.execute.metrics_correlation.MetricsCorrelationOutput;
+import org.opensearch.ml.engine.ExecuteException;
+import org.opensearch.ml.engine.MLEngine;
+import org.opensearch.ml.engine.MLExecutable;
+import org.opensearch.ml.engine.ModelHelper;
+import org.opensearch.ml.engine.algorithms.metrics_correlation.MetricsCorrelationTranslator;
+import org.opensearch.ml.engine.utils.ZipUtils;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.opensearch.ml.engine.ModelHelper.*;
+import static org.opensearch.ml.engine.utils.FileUtils.deleteFileQuietly;
+
+@Log4j2
+public abstract class DLModelExecute implements MLExecutable {
```

Comment:
This class contains a lot of duplicate code with https://github.com/opensearch-project/ml-commons/blob/2.x/ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/DLModel.java.

Is it possible to simplify the code? Like extend `DLModel`?

Reply:
Discussed offline, we will refactor this in next release.